test/system/libpod_suite_test.go

```diff
@@ -0,0 +1,696 @@
+		}
+	}
+
+	podmanBinary := "podman"
+	if os.Getenv("PODMAN_BINARY") != "" {
+		podmanBinary = os.Getenv("PODMAN_BINARY")
+	}
+
+	p := PodmanTest{
+		PodmanBinary:     podmanBinary,
+		ArtifactPath:     ARTIFACT_DIR,
+		TempDir:          tempDir,
+		GlobalOptions:    globalOptions,
+		PodmanCmdOptions: podmanCmdOptions,
+	}
+
+	// Setup registries.conf ENV variable
+	p.setDefaultRegistriesConfigEnv()
+	return p
+}
+
+//MakeOptions assembles all the podman main options
+func (p *PodmanTest) MakeOptions(args []string) []string {
+	var addOptions, subArgs []string
+	for _, n := range GLOBALOPTIONS {
+		if p.GlobalOptions[n] != "" {
+			addOptions = append(addOptions, n, p.GlobalOptions[n])
+		}
+	}
+
+	if len(args) == 0 {
+		return addOptions
+	}
+
+	subCmd := args[0]
+	addOptions = append(addOptions, subCmd)
+	if subCmd == "unmount" {
+		subCmd = "umount"
+	}
+	if subCmd == "help" {
+		subCmd = "h"
+	}
+
+	if _, ok := p.PodmanCmdOptions[subCmd]; ok {
+		m := make(map[string]bool)
+		subArgs = p.PodmanCmdOptions[subCmd]
+		for i := 0; i < len(subArgs); i++ {
+			m[subArgs[i]] = true
+		}
+		for i := 1; i < len(args); i++ {
+			if _, ok := m[args[i]]; !ok {
+				subArgs = append(subArgs, args[i])
+			}
+		}
+	} else {
+		subArgs = args[1:]
+	}
+
+	addOptions = append(addOptions, subArgs...)
+
+	return addOptions
+}
+
+// Podman is the exec call to podman on the filesystem, uid and gid the credentials to use
+func (p *PodmanTest) PodmanAsUser(args []string, uid, gid uint32, env []string) *PodmanSession {
+	podmanOptions := p.MakeOptions(args)
+	if os.Getenv("HOOK_OPTION") != "" {
+		podmanOptions = append(podmanOptions, os.Getenv("HOOK_OPTION"))
+	}
+	if env == nil {
+		fmt.Printf("Running: %s %s\n", p.PodmanBinary, strings.Join(podmanOptions, " "))
+	} else {
+		fmt.Printf("Running: (env: %v) %s %s\n", env, p.PodmanBinary, strings.Join(podmanOptions, " "))
+	}
+	command := exec.Command(p.PodmanBinary, podmanOptions...)
+
+	if uid != 0 || gid != 0 {
+		command.SysProcAttr = &syscall.SysProcAttr{}
+		command.SysProcAttr.Credential = &syscall.Credential{Uid: uid, Gid: gid}
+	}
+	if env != nil {
+		command.Env = env
+	}
+
+	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+	if err != nil {
+		Fail(fmt.Sprintf("unable to run podman command: %s\n%v", strings.Join(podmanOptions, " "), err))
+	}
+	return &PodmanSession{session}
+}
+
+// Podman is the exec call to podman on the filesystem
+func (p *PodmanTest) Podman(args []string) *PodmanSession {
+	return p.PodmanAsUser(args, 0, 0, nil)
+}
+
+//WaitForContainer waits on a started container
+func WaitForContainer(p *PodmanTest) bool {
+	for i := 0; i < 10; i++ {
+		if p.NumberOfRunningContainers() == 1 {
+			return true
+		}
+		time.Sleep(1 * time.Second)
+	}
+	return false
+}
+
+// Cleanup cleans up the temporary store
+func (p *PodmanTest) Cleanup() {
+	// Remove all containers
+	stopall := p.Podman([]string{"stop", "-a", "--timeout", "0"})
+	stopall.WaitWithDefaultTimeout()
```

Comment:
Actually the next step "rm" "-fa" will do this. "-f" means force. I guess here is make the kill more graceful.

Reply:
Yes, that is my understanding.  Issue here is with podman itself stop working or very slow because too many containers running.  Fall back to not-graceful kill, will help allow more testing rather than fail system-test after global timeout.  Often global system-test timeout is very long time (hours).

Also, from my experience, ``Cleanup()`` function should always exit clean, never any error stop testing.  (Maybe this is different in ginkgo).  For example:

```go
func (p *PodmanTest) Cleanup() {
    //make sure all cleanup steps always happen
    error_1 = stopall.WaitWithDefaultTimeout()
    error_2 = killPid(p.pid)
    error_3 = cleanupTmpDir()
    // show error, don't stop testing
    if error_1 or error_2 or error_3
        print(error)...
```